pkg/bot/command.go

```diff
@@ -96,94 +96,163 @@ func (b *Bot) handleCommand(bot *tgbotapi.BotAPI, update tgbotapi.Update, r *rad
 		b.DeleteMovieUserStates[userID] = command
 		b.UserActiveCommand[userID] = "DELETEMOVIE"
 		msg.Text = "Which movie would you like to delete?\n"
-		b.sendLibraryAsInlineKeyboard(movies, &msg)
+		b.sendMoviesAsInlineKeyboard(movies, &msg)
 
-	case "rss", "RSS":
-		command := radarr.CommandRequest{
-			Name:     "RssSync",
-			MovieIDs: []int64{},
+	case "up", "upcoming":
+		calendar := radarr.Calendar{
+			Start:       time.Now(),
+			End:         time.Now().AddDate(0, 0, 30), // 30 days
+			Unmonitored: *starr.True(),
 		}
-		_, err := r.SendCommand(&command)
+		upcoming, err := r.GetCalendar(calendar)
 		if err != nil {
 			msg.Text = err.Error()
 			fmt.Println(err)
 			b.sendMessage(msg)
 			break
 		}
-		msg.Text = "RSS sync started"
-		b.sendMessage(msg)
+		if len(upcoming) == 0 {
+			msg.Text = "no upcoming releases in the next 30 days"
+			msg.ParseMode = "MarkdownV2"
+			msg.DisableWebPagePreview = true
+			b.sendMessage(msg)
+			break
+		}
+		b.sendUpcoming(upcoming, &msg, bot)
 
-	case "wanted":
+	case "movies", "library":
 		movies, err := r.GetMovie(0)
 		if err != nil {
 			msg.Text = err.Error()
 			fmt.Println(err)
 			b.sendMessage(msg)
 			break
 		}
-		var monitoredMoviesIDs []int64
+		b.sendMovies(movies, &msg, bot)
+
+	case "ondisk":
+		movies, err := r.GetMovie(0)
+		var onDisk []*radarr.Movie
 		for _, movie := range movies {
-			if movie.Monitored == true {
-				monitoredMoviesIDs = append(monitoredMoviesIDs, movie.ID)
+			if movie.SizeOnDisk > 0 {
+				onDisk = append(onDisk, movie)
 			}
 		}
-		command := radarr.CommandRequest{
-			Name:     "MoviesSearch",
-			MovieIDs: monitoredMoviesIDs,
+		if err != nil {
+			msg.Text = err.Error()
+			fmt.Println(err)
+			b.sendMessage(msg)
+			break
+		}
+		b.sendMovies(onDisk, &msg, bot)
+
+	case "missing":
+		movies, err := r.GetMovie(0)
+		var missing []*radarr.Movie
+		for _, movie := range movies {
+			if movie.SizeOnDisk == 0 && movie.Monitored {
```

Comment:
Is this true? Even if it's currently downloading `SizeOnDisk` is `0`?

Reply:
hmm, maybe `HasFile` instead?